libs/ledger-live-common/src/onboarding/hooks/useOnboardingStatePolling.ts

```diff
@@ -0,0 +1,246 @@
+    });
+
+  const [fatalError, setFatalError] = useState<Error | null>(null);
+
+  useEffect(() => {
+    let onboardingStatePollingSubscription: Subscription;
+
+    if (device) {
+      console.log(
+        `SyncOnboarding: 🧑‍💻 new device: ${JSON.stringify(device)}`
+      );
+
+      onboardingStatePollingSubscription = onboardingStatePolling({
+        deviceId: device.deviceId,
+        pollingPeriodMs,
+      }).subscribe({
+        next: (onboardingStatePollingResult: OnboardingStatePollingResult) => {
+          console.log(
+            `SyncOnboarding: device version info ${JSON.stringify(
+              onboardingStatePollingResult
+            )}`
+          );
+          // FIXME: if null -> initialState ? What should be the initialOnboardingState ?
+          // Does not update the state if it could not be extracted from the flags
+          if (onboardingStatePollingResult) {
+            console.log("SETTING THE ONBOARDING STATE POLLING RESULT");
+            setOnboardingStatePollingResult(onboardingStatePollingResult);
+          }
+        },
+        error: (error) => {
+          console.log(
+            `SyncOnboarding: error ending polling ${error} -> ${JSON.stringify({
+              error,
+            })}`
+          );
+          setFatalError(
+            new DeviceOnboardingStatePollingError(
+              `Error from: ${error?.name ?? error} ${error?.message}`
+            )
+          );
+        },
+      });
+    }
+
+    return () => {
+      console.log("SyncOnboarding: cleaning up polling 🧹");
+      onboardingStatePollingSubscription?.unsubscribe();
+    };
+  }, [device, pollingPeriodMs, setOnboardingStatePollingResult]);
+
+  return { ...onboardingStatePollingResult, fatalError };
+};
+
+// TODO: Put in live-common/src/onboarding/onboardingStatePolling ?
+export const onboardingStatePolling = ({
+  deviceId,
+  pollingPeriodMs,
+}: {
+  deviceId: string;
+  pollingPeriodMs: number;
+}): Observable<OnboardingStatePollingResult> => {
+  console.log("🏎 GOING TO START");
+
+  // Could just be a boolean: firstRun ?
+  let i = 0;
+  // getDelayedOnboardingStateOnce
+  const getDelayedOnboardingStateOnce: Observable<OnboardingStatePollingResult> =
+    new Observable((subscriber) => {
+      console.log(`SyncOnboarding: ▶️ Polling from Observable ${i}`);
+      const delayMs = i > 0 ? pollingPeriodMs : 0;
+      console.log(`SyncOnboarding: polling delayed by ${delayMs} ms`);
+      i++;
+
+      const getOnboardingStateOnce =
+        (): Observable<OnboardingStatePollingResult> =>
+          withDevice(deviceId)((t) => from(getVersion(t))).pipe(
+            // TODO: choose timeout ms value. For now = polling period
+            timeout(pollingPeriodMs), // Throws a TimeoutError
+            first(),
+            catchError((error: any) => {
+              if (isAllowedOnboardingStatePollingError(error)) {
+                // Pushes the error to the next step to be processed (no retry from the beginning)
```

Comment:
You can maybe use partition and merge here. To have two different maps for handling errors and state.